nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -193,6 +194,21 @@ def __init__(self, cfg: MTEncDecModelConfig, trainer: Trainer = None):
             max_delta_length=cfg.max_generation_delta,
         )
 
+        # tie embedding weights
+        if self.shared_embeddings:
+            # validate vocabulary size and embedding dimension
+            if (
+                self.encoder.embedding.token_embedding.weight.shape
```

Comment:
Maybe better to also check the encoder and decoder use the same tokenizer (i.e. the `shared_tokenizer` is set)?

Reply:
Do we have a reliable method to compare tokenizers?
I was considering it but I am not not sure the "==" operator is working as we would like (i.e., comparing the class, all flags/options, and all tokens).
Even if we can compare the class correctly, not all classes allow comparison of the tokens.
Instead, I opted to leave it to the user to make sure the same tokenizers are being used.

Reply:
Oh, I just meant checking if the `shared_tokenizer` param in the config is set

Reply:
Isn't `shared_tokenizer` only affect learning a new tokenizer? I never changed the value of this parameter (defaults to true), even when using independent tokenizers.

Reply:
I guess its a fair assumption that all params are set correctly. In the worst case, the user can correct the param if they see an error of shared_tokenizers.
But I don't mind this staying as is either.

Reply:
Added an error if shared_tokenizer is False but shared_embeddings is True.